crates/bevy_render/src/shader/shader.rs

```diff
@@ -294,6 +295,51 @@ impl ShaderStages {
     }
 }
 
+/// Compute stages in a shader program
+#[derive(Clone, Debug)]
+pub struct ComputeShaderStages {
+    pub compute: Handle<Shader>,
+}
+
+pub struct ComputeShaderStagesIterator<'a> {
+    shader_stages: &'a ComputeShaderStages,
+    state: u32,
```

Comment:
since this can only yield a single item, having this be a `bool` would make more semantical sense.
Something like:

```
    pub struct ComputerShader... {
        // ...
        should_yield: false,
    }

impl<'a> Iterator for ... {
    fn next(&mut self) -> Option<Self::Item> {
        if self.should_yield{
            self.yielded = true;
            Some(self.shared_stages.compute.clone_weak());    
        } else {
           None
        }
    }
}
```